public/js/studentApplicationForm.js

```diff
@@ -13,16 +15,168 @@ nextButtons.forEach(($prevButton) => {
     $prevButton.addEventListener('click', handleClick)
 })
 
+hearAbout.addEventListener('change', handlePage4Change)
+
+
+function handlePage4Change(){
+    if(this[this.selectedIndex].text === 'Word of mouth'){
+        formWrappers[3].querySelector('#additionalNotesWordOfMouth').classList.remove('hidden')
+    }
+    else{
+        formWrappers[3].querySelector('#additionalNotesWordOfMouth').classList.add('hidden')
+    }
+    if(this[this.selectedIndex].text === 'Other'){
+        formWrappers[3].querySelector('#additionalNotesOther').classList.remove('hidden')
+    }
+    else{
+        formWrappers[3].querySelector('#additionalNotesOther').classList.add('hidden')
+    }
+}
+
+
 function handleClick(e){
     e.preventDefault()
+    let check = true
     formWrappers.forEach((formWrapper)=>{
-        if(formWrapper.id === e.currentTarget.value){
-            formWrapper.classList.remove('hidden')
-        }else{
-            formWrapper.classList.add('hidden')
+        if(formWrapper.id === e.currentTarget.dataset.page && e.currentTarget.className === 'nextButton'){
+            check = findPageValidation(parseInt(formWrapper.id), formWrapper)
         }
     })
-    pageCounter.textContent = e.currentTarget.value
+    if(check){
+        formWrappers.forEach((formWrapper)=>{
+            if(formWrapper.id === e.currentTarget.value && check){
+                formWrapper.classList.remove('hidden')
+            }else{
+                formWrapper.classList.add('hidden')
+            }
+        })
+        pageCounter.textContent = e.currentTarget.value
+    }
 }
 
+function findPageValidation(number, formWrapper){
+    let output = false
+    switch (number) {
+        case 4:
+            output = pageFourValidation(formWrapper)
+            break;
+        case 3:
+            output = pageThreeValidation(formWrapper)
+            break;
+        case 2:
+            output = pageTwoValidation(formWrapper)
+            break;
+        case 1:
+            output = pageOneValidation(formWrapper)
+            break;
+        default:
+            output = true
+    }
+    return output
+}
+
+function pageOneValidation(formWrapper) {
+    let inputs = formWrapper.querySelectorAll('input')
```

Comment:
could these first 6 lines be combined into a separate function and called in each page validator to make the code more DRY?